### PR TITLE
Handle language changes over websocket channel

### PR DESCRIPTION
**Pull request by atavism**
When switching languages, it's no longer necessary to switch tabs and the changes show up immediately

### CODE REVIEW

File: lib/common/session_model.dart [hunk 1]
```diff
@@ -183,9 +178,21 @@ class SessionModel extends Model {
...
       defaultValue: 'en',
+      onChanges: (setValue) {
+        if (websocket == null) return;
+        websocket.messageStream.listen(
+          (json) {
+            if (json["type"] != "pro") return;
+            final language = json["message"]["language"];
+            if (language != null && language != "") setValue(language);
+          },
+          onError: (error) => print(error),
```
Comment (jigar-f) [inline lib/common/session_model.dart:193]: Can we use appLogger here?

File: lib/common/session_model.dart [hunk 2]
```diff
@@ -183,9 +178,21 @@ class SessionModel extends Model {
...
     }
+    final websocket = WebsocketImpl.instance();
     return ffiValueBuilder<String>(
       'lang',
       defaultValue: 'en',
+      onChanges: (setValue) {
+        if (websocket == null) return;
+        websocket.messageStream.listen(
+          (json) {
+            if (json["type"] != "pro") return;
```
Comment (jigar-f) [inline lib/common/session_model.dart:189]: Not super important, but Do you think we can create a utils method for  `if (json["type"] != "pro") return;` check we are starting to use this at multiple places, In the future if the needed change we need to change only one?

**Review by jigar-f — APPROVED** [review]
LGTM, Just two small comments.

**Comment by jigar-f** [comment]
@atavism While you are socket changes, Could you add 

- If I go on the Pro code activation flow, If I enter code and get success but I still see an old UI, For a user to become a pro, the User has to restart the app.

File: lib/common/session_model.dart (hunk 2 above)
Comment (atavism) [reply lib/common/session_model.dart]: Done!

**Comment by atavism** [comment]
@jigar-f With this [change](https://github.com/getlantern/lantern-client/pull/1032/commits/d6c71facdf1c76013ead6f9e396c413afd6d3224), after successfully redeeming a reseller code, user data is immediately refreshed. I confirmed I get the Pro home screen

**Comment by atavism** [comment]
@jigar-f Hmm, just noticed the checkout page was disabled on desktop for some reason: https://github.com/getlantern/lantern-client/blob/main/lib/plans/plan_details.dart#L131

Just committed some changes to re-enable it.

File: lib/common/session_model.dart [hunk 3]
```diff
@@ -73,6 +73,13 @@ class SessionModel extends Model {
   late ValueNotifier<bool?> proxyAvailable;
   late ValueNotifier<String?> country;
 
+  // wsMessageProp parses the given json, checks if it represents a pro user message and
+  // returns the value (if any) in the map for the given property.
+  String? wsMessageProp(Map<String, dynamic> json, String field) {
```
Comment (jigar-f) [inline lib/common/session_model.dart:78]: Thanks for making this change.

**Comment by atavism** [comment]
Thanks for the review, @jigar-f!!

**Pull request merged**